scripts_matlab/module_SCONN.m

```diff
@@ -37,10 +37,14 @@
 for(nz=1:size(seedvol,3)) % dilate each slice
    %
    slc = seedvol(:,:,nz);
+   
    %LMP modified for octave
-   %slc = imdilate(slc,strel('disk',1));
-   slc = imdilate(slc,strel('disk',1,4));
-   %
+   if in_octave()
+       slc = imdilate(slc,strel('disk',1,0));
```

Comment:
Hi Mark,

I'd suggest creating a more generic `imdilate_generic()` function, and use that in place of imdilate, instead of copy-pasting if condition everywhere, to avoid different places going out of sync.

Reply:
Not sure what that would accomplish, since it is the creation of the an input parameter that has the issue.

Reply:
If ever changes had to be made to imdilate, or the way it is used, one change will propagate everywhere.

Reply:
I'm still not sure why you would want me to modify a standard builtin matlab/octave (imdilate) function?

Reply:
You misunderstood me - what I mean to suggest was a new auxilliary function in `imdilate_generic.m`:
with this code:
```matlab
function imdilate_generic()

   if in_octave()
       slc = imdilate(slc,strel('disk',1,0));
   else
       slc = imdilate(slc,strel('disk',1));
   end

end
```

and use imdilate_generic everywhere.

Reply:
this function is only used twice throughout OPPNI (inside loops) is it worth adding the processing overhead of a function call for 3 lines of code?

Reply:
Few milliseconds does not hold us back in anything, whereas subtle bugs in code can hold us back months sometimes with significant headache.

Reply:
Also, could you take review all the changes to see if a similar thing is repeated anywhere else?

Reply:
Discussed this code with Nathan, Aligning the number of line structuring element for Matlab and Octave should not have any impact, therefore the code can be modified to slc = imdilate(slc,strel('disk',1,0)) (no approximation) from the default of  4 within Matlab, to accommodate Octave. No need to specifically check for Octave.  I will make this change.

Reply:
Great. Could you take review all the changes to see if you do not DRY (do-not-repeat-yourself) anywhere else?